### ISSUE TITLE
Web2py 2.17.1 breaks backwards compatibility with Recaptcha

### ISSUE THREAD

**Issue by rdhakate** [web2py/web2py#2001]
**Describe the bug**
Recaptcha when used like in below statement does not work
`from gluon.tools import Recaptcha`

**To Reproduce**
1. Simply import Recaptcha as specified in description above and run the application.
2. We will get a ticket saysing **<type 'exceptions.NameError'> global name 'Recaptcha' is not defined**

**Expected behavior**
It should not throw any exception. With older version of web2py we get a message saying that you need to upgrade to Recaptch2. I expect the application to behave in a similar way with new version of web2py

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Win10
 - Browser  All

**Comment by BuhtigithuB** [comment]
Use from gluon.tools import Recaptcha2 as Recaptcha

If you can't or won't upgrade you code.

Recaptcha API was deprecated if I recall, that why we ditch the web2py
Recaptcha class... I had asked if we want to rename Recaptcha2 into
Recaptcha, but I have been busy and don't know what the response was... It
appears that the new release get out without this decision been made.

I think you can close this issue or try to make search in the web2py github
issue if there is not already a better one existing about the
Recaptcha/Rechaptcha2 issue or how the best way to make this change... Or
create a new issue targeting the proper problem which is the removal of the
Recaptcha class and might not been properly notified in change log that you
have to replace by Recaptcha2 which I doubt but I didn't check.

Thanks

**Comment by leonelcamara** [comment]
I also disagree with this being a web2py bug, it was not web2py who broke backwards compatibility it was google.

**Comment by rdhakate** [comment]
Agreed with the comments - but then, why is it that it displays an appropriate message when I run the application on web2py 2.15.x and not under web2py2.17.1 ? If you still think it is not an issue I would be glad to close it. 
Anyways I've now upgraded to Recaptcha2 so my issue is solved. Just let me know.

**Comment by leonelcamara** [comment]
I actually think the issue is having a recaptcha that seems to work but doesn't which was what happened with the last version, so I think the current behavior of breaking the old recaptcha user's apps should stay. However, I'm not the one who makes these decisions we have our BDFL for that.

**Comment by BuhtigithuB** [comment]
I think throw a warning in the web2py logs is legitimate though and is
actually a good idea and I would say a good practices that we are not
applying systematicaly... As such thing are generally only annonced (when
they are) over the mailing list and in the change log file... I think most
of the time when such API change occurs they are diffused over the mailing
list and the change log, but for sure sometimes they are only diffused in
one or the other mean... We we change the API we should maybe throw some
warning at startup or leave class there and throw warning in change log,
etc. There is many ways to do that. Projects are also annoncing deprecation
in advance in the Sphinx documentation oftenly... We could also have
shadowed the Recaptcha2 by the Recaptcha for the end user to maintain the
API... I would say that it partly my bad as I didn't persue this
modification till the change occurs in the smoothess way possible... It
wasn't so clear if it would be better to rename Recpatcha2 and silently
swap Recaptcha by Recaptcha2 or not as I am not aware of the impact of the
Recaptcha API deprecation, is all the Recaptcha feature are supported and
been transitioned properly while moving up the Recaptcha API version...
Will the web2py users rely on some of this feature available in Recaptcha
that is not supported anymore by Recaptcha2 and have hard time discover it
ceased functioning silently... Breaking the app the way we did appears to
me the cheapest way to move over... Sorry about that. We can decide what to
do, should we make sure we set a warning in the log to the user of the
Recaptcha class? If so we will reintroduce the class and make it log a
warning in web2py logs...

I can send a PR for it once it been determined it what we want to do... Let
me know??

**Comment by rdhakate** [comment]
Hey All,
     Glad to learn all the views  - Are we planning anything for this ? or should I close this ticket ? 

Rahul
